tests/test_processing_common.py

```diff
@@ -0,0 +1,127 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import json
+import tempfile
+import unittest
+
+from transformers import CLIPTokenizerFast, ProcessorMixin
+from transformers.models.auto.processing_auto import processor_class_from_name
+from transformers.testing_utils import (
+    check_json_file_has_correct_format,
+    require_tokenizers,
+    require_torch,
+    require_vision,
+)
+from transformers.utils import is_vision_available
+
+
+if is_vision_available():
+    from transformers import CLIPImageProcessor
+
+
+@require_torch
+class ProcessorTesterMixin:
+    processor_class = None
+
```

Comment:
missing as setup function for tempdirname that is pretty usefull

Reply:
setup function so far are in the subclasses (explicitly) of `ProcessorTesterMixin`. They have their own logic and are not in common.

Reply:
If you are talking just `self.tmpdirname = tempfile.mkdtemp()`, yes, I can move that to `ProcessorTesterMixin`

Reply:
yes was the only relevant one IMO

Reply:
OK, I will do it in a follow up PR that will update all processor test file to inherit from `ProcessorTesterMixin`.

Note: the current existing processor tests won't benefit from this new `ProcessorTesterMixin` --> only the new processors in the future that have extra arguments (to be saved) will

Reply:
Ok